File: WebClient SDK/src/PnrScripts/UI/PnrRegulationUIExtension.ts
import { CardUIExtension, ICardUIExtensionContext } from 'tessa/ui/cards';
import { Guid, createTypedField, DotNetType } from 'tessa/platform';
import { CardStoreMode } from 'tessa/cards';
import { DateTimeViewModel } from 'tessa/ui/cards/controls';
import { CardRequest, CardService } from 'tessa/cards/service';
import { PnrCardTypes } from '../Shared/PnrCardTypes';
import { PnrRequestTypes } from '../Shared/PnrRequestTypes';
import Moment from 'moment';

export class PnrRegulationUIExtension extends CardUIExtension {
  public async initialized(context: ICardUIExtensionContext) {
    if (!Guid.equals(context.card.typeId, PnrCardTypes.PnrRegulationTypeID)) {
      return;
    }
    // Создание карточки
    if (context.card.storeMode == CardStoreMode.Insert)
    {
      const regDateControl = context.model.controls.get('RegistrationDate') as DateTimeViewModel;
      if (!regDateControl) return;
      // Заполнить поле Дата регистрации текущей датой
      regDateControl.selectedDate = Moment(Date.now());

      // ЦФО(по Инициатору) при создании карточки
      let dci = context.card.sections.tryGet("DocumentCommonInfo");
      if (dci != null) {
        await this.SetUserDepartmentInfo(dci.fields.get('AuthorID'), context);
      }
    }
  }

  private async SetUserDepartmentInfo(authorID, context: ICardUIExtensionContext) {
    let CFOID;
    let CFOName;
    let request = new CardRequest();
        request.requestType = PnrRequestTypes.GetUserDepartmentInfoRequestTypeID;
        request.info["authorID"] = createTypedField(authorID, DotNetType.Guid);
    // проверка ответа
    let response = await CardService.instance.request(request);
    if (response.validationResult.isSuccessful) {
      CFOID = response.info.CFOID;
      CFOName = response.info.CFOName;
    }

    let PnrRegulations = context.card.sections.tryGet("PnrRegulations");
    if (CFOID != null && CFOName != null && PnrRegulations != null) {
      PnrRegulations.fields.set('CFOID', createTypedField(CFOID.$value, DotNetType.Guid))
      PnrRegulations.fields.set('CFOName', createTypedField(CFOName.$value, DotNetType.String))
    }
  }
}

File: Extensions/Tessa.Extensions.Client/UI/PnrRegulationUIExtension.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Tessa.Cards;
using Tessa.Extensions.Shared.PnrCards;
using Tessa.Platform.Storage;
using Tessa.UI;
using Tessa.UI.Cards;
using Tessa.UI.Cards.Controls;

namespace Tessa.Extensions.Client.UI
{
    public sealed class PnrRegulationUIExtension : CardUIExtension
    {
        private readonly ICardRepository cardRepository;
        public PnrRegulationUIExtension(ICardRepository cardRepository)
        {
            this.cardRepository = cardRepository;
        }

        private async void SetUserDepartmentInfo(Guid? authorID, ICardModel cardModel)
        {
            CardRequest request = new CardRequest
            {
                RequestType = Shared.PnrRequestTypes.GetUserDepartmentInfoRequestTypeID,
                Info =
                {
                    { "authorID", authorID }
                }
            };

            CardResponse response = await cardRepository.RequestAsync(request);

            Tessa.Platform.Validation.ValidationResult result = response.ValidationResult.Build();
            TessaDialog.ShowNotEmpty(result);
            if (result.IsSuccessful)
            {
                cardModel.Card.Sections["PnrRegulations"].Fields["CFOID"] = response.Info.Get<Guid?>("CFOID");
                cardModel.Card.Sections["PnrRegulations"].Fields["CFOName"] = response.Info.Get<string>("CFOName");
            }
        }

        public override Task Initialized(ICardUIExtensionContext context)
        {
            Card card = context.Card;
            ICardModel cardModel = context.Model;

            if(card.StoreMode == CardStoreMode.Insert)
            {
                // Заполнить поле Дата регистрации текущей датой
                if (cardModel.Controls.TryGet("RegistrationDate", out IControlViewModel regDateControl))
                {
                    ((DateTimeViewModel)regDateControl).SelectedDate = DateTime.Now;
                }
                // ЦФО(по Инициатору) при создании карточки
                SetUserDepartmentInfo((Guid)card.Sections["DocumentCommonInfo"].Fields["AuthorID"], cardModel);
            }

            return base.Initialized(context);
        }
    }
}
